tfsdk/serve_config.go

```diff
@@ -0,0 +1,28 @@
+package tfsdk
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-go/tfprotov6"
+)
+
+// parseConfig turns a *tfprotov6.DynamicValue and a Schema into a Config.
+func parseConfig(ctx context.Context, dv *tfprotov6.DynamicValue, schema Schema) (ReadOnlyData, diag.Diagnostics) {
```

Comment:
Drive-by nit: I'd be tempted to name this `newReadOnlyData` for clarity, unless its updated to return a `Config` (which I'd still opt for the new/New prefix).